src/server/routes/entity/entity.js

```diff
@@ -482,9 +485,9 @@ async function getNextAnnotation(
 		id && new Annotation({id}).fetch({transacting})
 	);
 
-	return orm.func.annotation.updateAnnotation(
+	return body.annotation ? orm.func.annotation.updateAnnotation(
 		orm, transacting, oldAnnotation, body.annotation, revision
-	);
+	) : Promise.resolve(null);
 }
 
```

Comment:
I did this because earlier if oldAnnotaion and newAnnotation both are undefined, updateAnnotation was thinking both are same and was just returning oldAnnotation

Reply:
I think this is absolutely fine and a justified optimization